src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -682,6 +682,19 @@ protected virtual void Dispose(bool disposing)
 
             if (disposing)
             {
+                if (this.executors != null)
+                {
+                    foreach (ProcessingSourceExecutor executor in this.executors)
+                    {
+                        if (executor.Processor is null)
```

Comment:
I did what you propose first, but that has a much greater impact on WPA. When we move WPA onto the Engine, we can more easily make that change.

Yes, ProcessingSourceReference can also calls dispose, but it is not owned by the Engine and shouldn't cause any issues.

Seeing as the executor is going away immediately in this case, I don't see a reason to set it to null.

Happy to chat more offline sometime about some of the history behind this.

Reply:
Just to be clear I wasn't indicating that `ProcessingSourceReference` also doing disposal would cause any issue, but it has an extra step when disposing processors, which is different from what the engine is doing here

Reply:
Actually, I thought about it overnight and I think it would be ok to go back to implementing IDisposable on ProcessingSourceExecutor. As WPA doesn't need to dispose of this, it can continue to do its own thing.

Reply:
I'll do this in another commit as I'd like to get this change in to unblock work.

Reply:
Thanks again @xj-ms. I think the new update should address these issues.